File: FurnitureStore/FurnitureStore.Models/Category/Category.cs

namespace FurnitureStore.Models.Category
{
    public class Category
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public long? ImageId { get; set; }
        public string? ImagePath { get; set; }
    }
}

File: FurnitureStore/UI/furniturestore_admin/lib/models/category/category.dart
import 'package:json_annotation/json_annotation.dart';

part 'category.g.dart';

@JsonSerializable()
class CategoryModel {
  int? id;
  String? name;
  String? description;
  String? imagePath;
  int? imageId;

  CategoryModel(
      this.id, this.name, this.description, this.imagePath, this.imageId);

  factory CategoryModel.fromJson(Map<String, dynamic> json) =>
      _$CategoryModelFromJson(json);

  Map<String, dynamic> toJson() => _$CategoryModelToJson(this);
}
